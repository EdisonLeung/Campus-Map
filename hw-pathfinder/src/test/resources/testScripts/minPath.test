# Tests pathfinding for path with least cost but not least jumps

# Create Graph
CreateGraph minPath

# Add nodes
AddNode minPath a
AddNode minPath b
AddNode minPath c

# Add edges
AddEdge minPath a b 5.0
AddEdge minPath b c 5.0
AddEdge minPath a c 20.0

# test pathfinding
FindPath minPath a c