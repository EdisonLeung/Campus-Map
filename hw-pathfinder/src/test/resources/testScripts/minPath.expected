# Tests pathfinding for path with least cost but not least jumps

# Create Graph
created graph minPath

# Add nodes
added node a to minPath
added node b to minPath
added node c to minPath

# Add edges
added edge 5.000 from a to b in minPath
added edge 5.000 from b to c in minPath
added edge 20.000 from a to c in minPath

# test pathfinding
path from a to c:
a to b with weight 5.000
b to c with weight 5.000
total cost: 10.000